cluster-api/providers/openstack/go.mod

```diff
@@ -1,24 +1,27 @@
 module openshift/installer/cluster-api/providers/openstack
 
-go 1.20
+go 1.21
```

Comment:
the Installer folks seems to have a guideline on splitting vendoring commits.
One commit with `go.mod` and `go.sum` change and another for the vendor update, like we did in this PR https://github.com/openshift/installer/pull/7939/commits
context https://github.com/openshift/installer/pull/7939#discussion_r1494797147